ogr/ogrsf_frmts/geojson/ogrgeojsonreader.cpp

```diff
@@ -1934,7 +1934,13 @@ void OGRGeoJSONReaderAddOrUpdateField(
             if (eSubType == OFSTNone)
             {
                 if (eNewType == OFTStringList)
+                {
                     poFDefn->SetType(OFTStringList);
+                }
+                else if (eNewType == OFTInteger)
```

Comment:
what about OFTInteger64, OFTReal, and probably other types ?
Shouldn't that be:
```suggestion
                else if (eNewType != OFTString)
```

Reply:
Correct, test case added for real.